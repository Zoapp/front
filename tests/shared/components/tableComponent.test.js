import React from 'react';
import renderer from 'react-test-renderer';
import TableComponent from '@shared/components/tableComponent';

describe('TableComponent', () => {
  const newItem = ({ id, name, values = [], icon = null }) => {
    return {
      id: id || name,
      name,
      values,
      icon,
    };
  };

  it('renders correctly', () => {
    const headers = [];
    const items = [];

    const component = renderer.create(
      <TableComponent
        headers={headers}
        items={items}
        title="some title"
        onSelect={jest.fn()}
        selectedItem={0}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders items', () => {
    const headers = ['header 1'];
    const items = [
      newItem({ name: 'item 1' }),
    ];

    const component = renderer.create(
      <TableComponent
        headers={headers}
        items={items}
        title="some title"
        onSelect={jest.fn()}
        selectedItem={0}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders items with numerical ids', () => {
    const headers = ['header 1'];
    const items = [
      newItem({ id: 123, name: 'item 1' }),
    ];

    const component = renderer.create(
      <TableComponent
        headers={headers}
        items={items}
        title="some title"
        onSelect={jest.fn()}
        selectedItem={0}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
