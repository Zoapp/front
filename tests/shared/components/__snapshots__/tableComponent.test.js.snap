// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableComponent renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "minHeight": "400px",
      }
    }
  >
    some title
    <table
      className={null}
      style={
        Object {
          "borderSpacing": "0",
          "width": "100%",
        }
      }
    >
      <thead>
        <tr
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.26)",
              "fontSize": "13px",
              "height": "56px",
              "padding": "0 56px 0 0",
              "textAlign": "left",
              "verticalAlign": "middle",
            }
          }
        />
      </thead>
      <tbody />
    </table>
  </div>
  <div
    style={
      Object {
        "borderTop": "1px solid rgba(0, 0, 0, 0.12)",
        "height": "56px",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`TableComponent renders items 1`] = `
<div>
  <div
    style={
      Object {
        "minHeight": "400px",
      }
    }
  >
    some title
    <table
      className={null}
      style={
        Object {
          "borderSpacing": "0",
          "width": "100%",
        }
      }
    >
      <thead>
        <tr
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.26)",
              "fontSize": "13px",
              "height": "56px",
              "padding": "0 56px 0 0",
              "textAlign": "left",
              "verticalAlign": "middle",
            }
          }
        >
          <th
            style={
              Object {
                "paddingLeft": "24px",
                "textAlign": "left",
              }
            }
          >
            header 1
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="selectedListItem"
          onClick={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": "13px",
            }
          }
        >
          <td
            style={
              Object {
                "borderTop": "1px solid rgba(0, 0, 0, 0.12)",
                "height": "48px",
                "verticalAlign": "middle",
                "width": "48px",
              }
            }
          >
            
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    style={
      Object {
        "borderTop": "1px solid rgba(0, 0, 0, 0.12)",
        "height": "56px",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`TableComponent renders items with numerical ids 1`] = `
<div>
  <div
    style={
      Object {
        "minHeight": "400px",
      }
    }
  >
    some title
    <table
      className={null}
      style={
        Object {
          "borderSpacing": "0",
          "width": "100%",
        }
      }
    >
      <thead>
        <tr
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.26)",
              "fontSize": "13px",
              "height": "56px",
              "padding": "0 56px 0 0",
              "textAlign": "left",
              "verticalAlign": "middle",
            }
          }
        >
          <th
            style={
              Object {
                "paddingLeft": "24px",
                "textAlign": "left",
              }
            }
          >
            header 1
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="selectedListItem"
          onClick={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": "13px",
            }
          }
        >
          <td
            style={
              Object {
                "borderTop": "1px solid rgba(0, 0, 0, 0.12)",
                "height": "48px",
                "verticalAlign": "middle",
                "width": "48px",
              }
            }
          >
            
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    style={
      Object {
        "borderTop": "1px solid rgba(0, 0, 0, 0.12)",
        "height": "56px",
        "width": "100%",
      }
    }
  />
</div>
`;
